### PR TITLE
PP-6994 Use expunged refunds from ledger when calculating refundability

### PR DESCRIPTION
**Pull request by stephencdaly**
If the charge is historic, make a request to ledger to get any refunds that exist there. If the refunds also exist in the connector database, prefer the record from the database as the refund may be in-flight.

### CODE REVIEW

File: src/test/java/uk/gov/pay/connector/it/resources/sandbox/SandboxRefundsResourceIT.java [hunk 1]
```diff
@@ -286,22 +293,87 @@ public void shouldFailRequestingARefund_whenAPartialRefundMakesTotalRefundedAmou
         assertRefundsHistoryInOrderInDBForSuccessfulOrPartialRefund(defaultTestCharge);
     }
 
+    @Test
+    public void shouldFailRequestingARefundForHistoricCharge_whenPartialRefundAmountGreaterThanRemainingAmount_whenExistingPartialRefundsHaveBeenExpunged() throws Exception {
```
Comment (sfount) [inline src/test/java/uk/gov/pay/connector/it/resources/sandbox/SandboxRefundsResourceIT.java:297]: 😮

File: src/test/java/uk/gov/pay/connector/service/RefundServiceTest.java [hunk 2]
```diff
@@ -616,6 +640,55 @@ public void shouldFindRefundsGivenValidCharge() {
...
+        
+        var refundTransactionsForPayment = aValidRefundTransactionsForPayment()
+                .withParentTransactionId(charge.getExternalId())
+                .withTransactions(List.of(ledgerRefund1, ledgerRefund2))
+                .build();
+        when(mockLedgerService.getRefundsForPayment(charge.getGatewayAccountId(), charge.getExternalId()))
+                .thenReturn(refundTransactionsForPayment);
+
+        List<Refund> refunds = refundService.findRefunds(charge);
+        assertThat(refunds, hasSize(3));
```
Comment (sfount) [inline src/test/java/uk/gov/pay/connector/service/RefundServiceTest.java:689]: Would it be useful to add an assertion here to assert that the in-flight refund is always selected? It looks like the `refundInDatabaseAndLedgerId` have different mocked statuses so potentially an assertion on the status of that entry returned.

**Review by sfount — DISMISSED** [review]
Looks great to me 👍  I think it would be good to also add ledger/ connector db  path scenarios to the integration tests of critical code paths that use this service but that would be ideal as a separate PR

File: src/test/java/uk/gov/pay/connector/service/RefundServiceTest.java (hunk 2 above)
Comment (stephencdaly) [reply src/test/java/uk/gov/pay/connector/service/RefundServiceTest.java:689]: ah yeh good spot, I intended in those assertions but then must have got distracted

Comment (stephencdaly) [reply src/test/java/uk/gov/pay/connector/service/RefundServiceTest.java:689]: added in those assertions

**Review by sfount — APPROVED** [review]

**Pull request merged**